### PR TITLE
[NEW] - Mautic new connection option with Username and Password added.

### PR DESCRIPTION
**Pull request by nileshc-bsf**
- Mautic integration with Username and Password.

### CODE REVIEW

File: classes/class-bsf-cm-auto-update.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			return self::$instance;
+		}
+
+		/**
+		 * Constructor
+		 */
+		public function __construct() {
+
+			// Plugin Updates.
+			add_action( 'init', __CLASS__ . '::init' );
```
Comment (premanshup) [inline classes/class-bsf-cm-auto-update.php]: @nileshc-bsf Should this be called on `admin_init`?

File: contacts-in-mautic.php [hunk 2]
```diff
@@ -33,6 +33,12 @@
 if ( ! defined( 'WPINC' ) ) {
 	die;
 }
+// Set the BSF_CONTACT_MAUTIC_VERSION.
+define('BSF_CONTACT_MAUTIC_VERSION', '1.0.3');
```
Comment (premanshup) [inline contacts-in-mautic.php]: @nileshc-bsf This version is different than the version defined in comments in main plugin file - https://cl.ly/f535fe63c6d1 - Please check!

File: classes/class-bsf-cm-auto-update.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		}
+
+}
+
+endif;
+
+/**
+ * calling 'get_instance()' method
+ */
+Bsf_CM_Auto_Update::get_instance();
```
Comment (premanshup) [inline classes/class-bsf-cm-auto-update.php]: @nileshc-bsf We can replace `get_instance()` with simply creating an object of the class and initializing everything in constructor only.

File: contacts-in-mautic.php [hunk 4]
```diff
@@ -71,40 +83,54 @@ function bsf_mautic_cnt_scode( $bsf_atts ) {
...
-			$url      = $credentials['baseUrl'] . '/api/contacts?access_token=' . $access_token;
-		}
-		else {
-			$url      = $credentials['baseUrl'] . '/api/contacts?search=!is:anonymous&access_token=' . $access_token;
-		}
-		$response = wp_remote_get( $url );
-		if ( ! is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) === 200 ) {
-			$response_body    = $response['body'];
-			$contacts_details = json_decode( $response_body );
+	if ( 'mautic_user_pass' == $get_mautic_connect_type ) {
```
Comment (premanshup) [inline contacts-in-mautic.php]: @nileshc-bsf Please use strict type check.

Comment (nileshc-bsf) [reply contacts-in-mautic.php]: Strict type checked.

File: classes/class-bsf-cm-auto-update.php (hunk 3 above)
Comment (nileshc-bsf) [reply classes/class-bsf-cm-auto-update.php]: Removed the get_instance() and simply created an object of the class.

File: contacts-in-mautic.php (hunk 2 above)
Comment (nileshc-bsf) [reply contacts-in-mautic.php]: Version number updated to 1.0.2

File: classes/class-bsf-cm-auto-update.php (hunk 1 above)
Comment (nileshc-bsf) [reply classes/class-bsf-cm-auto-update.php]: Action changes to "admin_init".

**Pull request merged**